viewer-prototype/src/browser/trace-viewer/components/utils/filtrer-tree/utils.tsx

```diff
@@ -11,17 +11,26 @@ const entryToTreeNode = (entry: Entry) => ({
 export const listToTree = (list: Entry[]): TreeNode[] => {
     const rootNodes: TreeNode[] = [];
     const lookup: { [key: string]: TreeNode } = {};
+    // Fill in the lookup table
     list.forEach(entry => {
         lookup[entry.id] = entryToTreeNode(entry);
     });
-    Object.keys(lookup).forEach(id => {
-        const entry = lookup[id];
-        if (entry.parentId === -1) {
-            entry.isRoot = true;
-            rootNodes.push(entry);
-        } else if (entry.parentId in lookup) {
-            const p = lookup[entry.parentId];
-            p.children.push(entry);
+
+    // Create the tree in the order it has been e
+    list.forEach(entry => {
+        const node = entryToTreeNode(entry);
```

Comment:
const node = lookup[entry.id] ? entryToTreeNode has already been called on line 15

It's causing the hierarchy to fail since you're adding the child to the parent in the lookup list, but the root node you are adding is the one just created here.

Reply:
I'll update it. BTW, I have noticed that there is some issues for data providers in Experiments, e.g. experiment with 2 kernel traces, where in that case the Thread status or Resource status view only shows one trace. I'm looking into fixes in the Trace Compass server for that.